From a.beregalov@gmail.com  Fri Jul 10 15:23:01 2009
From: Alexander Beregalov <a.beregalov@gmail.com>
Date: Fri,  3 Jul 2009 16:25:50 +0400
Subject: Staging: at76_usb: fix !CONFIG_WIRELESS_EXT build
To: gregkh@suse.de, linux-kernel@vger.kernel.org
Cc: Alexander Beregalov <a.beregalov@gmail.com>
Message-ID: <1246623953-23215-1-git-send-email-a.beregalov@gmail.com>


Fix this build error when CONFIG_WIRELESS_EXT is not set
at76_usb.c:5322: error: 'struct net_device' has no member named 'wireless_data'
at76_usb.c:5325: error: 'struct net_device' has no member named 'wireless_handlers'

Signed-off-by: Alexander Beregalov <a.beregalov@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/at76_usb/at76_usb.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/drivers/staging/at76_usb/at76_usb.c
+++ b/drivers/staging/at76_usb/at76_usb.c
@@ -5319,10 +5319,12 @@ static int at76_init_new_device(struct a
 
 	/* Add pointers to enable iwspy support. */
 	priv->wireless_data.spy_data = &priv->spy_data;
+#ifdef CONFIG_WIRELESS_EXT
 	netdev->wireless_data = &priv->wireless_data;
+	netdev->wireless_handlers = &at76_handler_def;
+#endif
 
 	netdev->watchdog_timeo = 2 * HZ;
-	netdev->wireless_handlers = &at76_handler_def;
 	dev_alloc_name(netdev, "wlan%d");
 
 	ret = register_netdev(priv->netdev);
